cpp/oneapi/dal/backend/primitives/lapack/eigen.cpp

```diff
@@ -0,0 +1,98 @@
+/*******************************************************************************
+* Copyright 2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include <src/externals/service_lapack.h>
+
+#include "oneapi/dal/backend/primitives/lapack/eigen.hpp"
+#include "oneapi/dal/backend/dispatcher.hpp"
+#include "oneapi/dal/backend/interop/common.hpp"
+
+namespace oneapi::dal::backend::primitives {
+
+template <typename Float>
+static void syevd(char jobz,
```

Comment:
Just to double-check: sometimes we are talking about support for different SYCL backends for oneDAL. Don't we want to create primitives in a way it will be easily to use oneMKL underneath?
For example, here we are using host-only MKL small libs, but can possibly use oneMKL interfaces.

Reply:
Currently we need to call LAPACK on CPU. 

> Don't we want to create primitives in a way it will be easily to use oneMKL underneath?

In general, yes. But let's move step-by-step. It's easier to add universal DPC++ versions once GPU support is needed. 

The problem is that the small version of oneMKL we are using is GPU-only, there is no CPU branch underneath, so device dispatching should be controlled on our side.